### PR TITLE
Add support for the OpenType MATH table.

### PR DESCRIPTION
**Pull request by fred-wang**

### CODE REVIEW

File: src/hb-ot-layout.cc [hunk 1]
```diff
@@ -199,6 +206,21 @@ _get_gpos (hb_face_t *face)
...
 }
+static inline const OT::MATH&
+_get_math (hb_face_t *face)
+{
+  if (unlikely (!hb_ot_shaper_face_data_ensure (face))) return OT::Null(OT::MATH);
+
+  hb_ot_layout_t * layout = hb_ot_layout_from_face (face);
+
+  // If the MATH table is not loaded yet, do it now.
+  if (!layout->math_blob) {
```
Comment (behdad) [inline src/hb-ot-layout.cc]: This is not threadsafe.  You need to use `hb_atomic_ptr_get()` and `hb_atomic_ptr_cmpexch()` instead.

Comment (behdad) [inline src/hb-ot-layout.cc]: Also, do the atomic check on `layout->math`, not `layout->math_blob`.

File: src/hb-ot-layout.h [hunk 2]
```diff
@@ -297,6 +298,12 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
 			      unsigned int *range_end          /* OUT.  May be NULL */);
 
 
+/*
+ * MATH
+ */
+
+HB_EXTERN hb_bool_t
+hb_ot_layout_has_math_data (hb_face_t *face);
```
Comment (behdad) [inline src/hb-ot-layout.h]: Add two newlines after.

File: test/api/Makefile.am [hunk 3]
```diff
@@ -43,10 +43,22 @@ endif
...
-endif
+
+if HAVE_FREETYPE
+TEST_PROGS += \
+	test-ot-layout-math \
+	$(NULL)
+test_ot_layout_math_LDADD = $(LDADD)
+test_ot_layout_math_CPPFLAGS = $(AM_CPPFLAGS)
+test_ot_layout_math_CPPFLAGS += $(FREETYPE_CFLAGS)
+test_ot_layout_math_LDADD += $(FREETYPE_LIBS)
```
Comment (behdad) [inline test/api/Makefile.am]: Merge these four lines into two.

**Comment by behdad** [comment]
This is mostly merged into my math branch now.

**Comment by fred-wang** [comment]
It would be great to rebase d776da2f678e1895ac61ed9dfba10efc6d1c82ee too later, using the new API for mathvariant. http://frederic-wang.fr/opentype-math-in-harfbuzz.html

**Comment by fred-wang** [comment]
See also https://github.com/fred-wang/harfbuzz-tutorial

**Pull request closed without merging**